js/app.js

```diff
@@ -33,39 +35,56 @@ class Player {
     }
     restart(posX, posY) {
         if ((this.x >= posX - 50 && this.x <= posX + 50) && (this.y >= posY - 50 && this.y <= posY + 50)) {
-            this.x = 200;
-            this.y = 400;
+            document.getElementById('score').textContent = score;
+            document.getElementById('game-over').classList.remove('invisible');
+            document.getElementById('game-over').classList.add('visible');
+            gameStatus = 'pause';
+            score = 0;
         }
     }
     handleInput(key) {
         switch (key) {
             case 'left': {
-                if (this.x > 0) {
+                if (this.x > 0 && gameStatus === 'start') {
                     this.x -= 100;
                 }
                 break;
             }
             case 'up': {
-                if (this.y > 0) {
+                if (this.y > 0 && gameStatus === 'start') {
                     this.y -= 100;
+                    player.update_score();
                 }
                 break;
             }
             case 'right': {
-                if (this.x < 400) {
+                if (this.x < 400 && gameStatus === 'start') {
                     this.x += 100;
                 }
                 break;
             }
             case 'down': {
-                if (this.y < 400) {
+                if (this.y < 400 && gameStatus === 'start') {
                     this.y += 100;
+                    player.update_score();
                 }
                 break;
             }
+            case 'space-bar': {
+                gameStatus = 'start';
+                this.x = 200;
+                this.y = 400;
+                document.getElementById('game-over').classList.remove('visible');
```

Comment:
if for some reason the classList property returns undefined this code will fail because undefined.remove will throw a run-time error, so you should check for the classList to return the desired value before using remove on it.